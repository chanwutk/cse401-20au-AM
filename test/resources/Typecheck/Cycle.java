class Cycle {
	public static void main(String[] args) {
		System.out.println(new C1().f(0));
	}
}

class C1 extends C2 {
	public int f(int x) {
		return x;
	}
}

class C2 extends C1 {
	public int f(int x) {
		return x;
	}
}
