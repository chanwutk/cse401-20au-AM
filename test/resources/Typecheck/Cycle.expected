Cycle.java:7: error: cyclic inheritance involving C1
